nixos/modules/virtualisation/docker.nix

```diff
@@ -175,7 +175,8 @@ in
 
       systemd.services.docker = {
         wantedBy = optional cfg.enableOnBoot "multi-user.target";
-        after = [ "network.target" "docker.socket" ];
+        after = [ "network.target" "docker.socket" "containerd.service" ];
+        wants = [ "containerd.service" ];
```

Comment:
This is unnecessary, Docker manages its own containerd instance (unless that has recently/silently changed).

Reply:
Yeah, that split was only done upstream in the next major version upgrade and for other distros a longer time.